element-hq/element-x-android
# Hide join call button when the user is already in the call

## Content


When the user is in the call due to the Pip mode, the "Join" call button is displayed. This PR hides it if the user is currently (and locally) in the call.

This PR also improve the `RoomCallState` regarding permission. We now have a disabled button if the user is not allowed to join the call.

## Motivation and context


Closes https://github.com/element-hq/element-meta/issues/2608

## Screenshots / GIFs


See the recorded ones.

## Tests



- In a room, start a call
- Enter Pip mode
- Observe that the button to join the call in the room (top end of the screen) is not displayed, and it is also not displayed anymore in the timeline item of the call.

Note that if there is a call in another room, joining the other call will do nothing. This is another issue, not related to this PR.

## Tested devices

- OS version(s):

## Checklist
